### ISSUE TITLE
Crash under Wayland after upgrading to Ubuntu 22.04

### ISSUE THREAD

**Issue by pdroalves** [pdfpc/pdfpc#678]
**Expected Behaviour**
I'm loading a PDF presentation, which I used in the past with pdfpc, and I expect it to open.

**Actual Behaviour**
I receive a segmentation fault error and everything crash in one or two seconds.

```
➜  ~ pdfpc ~/Downloads/Defesa_de_doutorado___Apresenta__o-1.pdf
Screensaver inhibited
[1]    224099 segmentation fault (core dumped)  pdfpc ~/Downloads/Defesa_de_doutorado___Apresenta__o-1.pdf
```

**Steps to reproduce**

1. Install Ubuntu 20.04
2. Update to Ubuntu 22.04
3. Try to open any presentation using pdfpc 

pdfpc version: v4.6 or v4.5.1pre
Used Distribution (GTK version, vala version): Ubuntu 22.04
Display Server (X/Wayland): Wayland
HiDPI Setting (yes/no): no

**Comment by fnevgeny** [comment]
> I receive a segmentation fault error and everything crash in one or two seconds.

Without doing anything (pressing a key etc)?

>     1. Install Ubuntu 20.04
> 
>     2. Update to Ubuntu 22.04

Do you claim the problem is caused specifically by the upgrade? Installing 22.04 afresh is not enough?
 
>     3. Try to open any presentation using pdfpc

Including the [official demo](https://github.com/pdfpc/pdfpc/releases/latest/download/pdfpc-demo.pdf)?

**Comment by pdroalves** [comment]
Hi @fnevgeny 

Yes, the crash happens without pressing anything. I simply execute it from the command line without any parameter, it opens and shows my presentations for one second, and then crashes. This also happens with the official demo.

With me the problem started after upgrading, but I haven't tested with a fresh install. I will try it with a fresh installation of 22.04 on a VM to be sure.

**Comment by mvube** [comment]
I am having the same problem.
"segmentation fault (core dumped)"
Ubuntu 22.04
pdfpc version 4.5.0-3
Display: Wayland

**Comment by AndreasBilke** [comment]
What happens if you build pdfpc from source? Does the same happen? And what does the stack trace looks like?

**Comment by mvube** [comment]
Hi Andreas,

I will try that tomorrow and let you know.
How do I do a stack trace?

Pat

**Comment by AndreasBilke** [comment]
Just execute the builded pdfpc file. It should have debug symbols (with should result in stack trace) enabled by default.

**Comment by fnevgeny** [comment]
Well, I've just upgraded my home PC from 20.04 to 22.04 and see no problem running pdfpc, either with the old binary compiled under 20.04 or the recompiled one.

The only issue is a GDK warning message when running under Wayland (need to understand why) but no crash nevertheless.

@mvube, see if running `env GDK_BACKEND=x11 pdfpc ...` makes any change.

**Comment by mvube** [comment]
Success.  Thank you all.  I downloaded and installed the .zip package.
I found that I had to install a heap of .dev files to make it work.
Basically in short I ran the cmake command and every time an error showed up about a missing package I installed the required .dev package via Synaptic (I still love Synaptic) 

Now I have it running well in Wayland on Ubuntu 22.04  :-)

**Comment by fnevgeny** [comment]
> I found that I had to install a heap of .dev files to make it work. Basically in short I ran the cmake command and every time an error showed up about a missing package I installed the required .dev package via Synaptic (I still love Synaptic)

Just to note, there is a more straightforward (though perhaps less entertaining...) approach, namely, to follow the README and install all the prerequisites in one `apt` go.

> Now I have it running well in Wayland on Ubuntu 22.04 :-)

Wait. Do you say pdfpc from the Ubuntu package crashes while the one you compiled doesn't - on the same system?? Note that you may have both installed simultaneously, if desired - `/usr/bin/pdfpc` vs. `/usr/local/bin/pdfpc`.

**Comment by fnevgeny** [comment]
There is one unrelated thing. I've just noticed that the Ubuntu binary reports itself as 4.5.1*pre* (I saw it in OP's report but thought he was compiling from the respective git branch). IMO, this is confusing. It's clear that distributions may apply various patches, but the baseline should be clearly identified - in this case, 4.5.0, instead of pointing to a moving target. @barak, can you please comment?

**Comment by barak** [comment]
The Debian package is at 4.6.0-1. I don't believe Debian ever had 4.5.1, it went straight from 4.5.0 to 4.6.0.

The current version seems to work for me, on Debian testing with Gnome/Wayland. Can anyone reproduce this using the Debian package?

**Comment by fnevgeny** [comment]
> I don't believe Debian ever had 4.5.1

Right, the package version is 4.5.0, but the binary says (`pdfpc -v`) 4.5.1pre.

**Comment by barak** [comment]
Gotcha: I must have cherry-picked some post-4.5.0 commits and not bothered editing that out.

Does the bug manifest in 4.6.0-1? Because it seems to work on Wayland for me.

Should I add some extra stuff to the dependencies, perhaps?

**Comment by pdroalves** [comment]
Hello all!
I've just tried the apt version of pdfpc in a fresh Ubuntu 22.04 VM and had no issues. 

> @mvube, see if running `env GDK_BACKEND=x11 pdfpc ...` makes any change.

Tried this on my main machine and it works. By setting x11 to GDK_BACKEND, pdfpc opens without a crash. I have no idea why and wasn't able to get clues by compiling it.

**Comment by fnevgeny** [comment]
> I've just tried the apt version of pdfpc in a fresh Ubuntu 22.04 VM and had no issues.

Just to make sure, the new installation uses Wayland (`echo $XDG_SESSION_TYPE`), right? BTW, in your original report, you said the display server is X11, not Wayland...

> Tried this on my main machine and it works. By setting x11 to GDK_BACKEND, pdfpc opens without a crash.

Can you try a clean reinstall of the pdfpc apt package (`apt purge pdf-presenter-console && apt install pdf-presenter-console`) on your main PC and see if it makes any difference?

**Comment by pdroalves** [comment]
The VM installation uses Wayland, **as well as** my main PC. Sorry, I misunderstood the form for the bug report. When I wrote "X" was like marking X on a checkbox asking if I was using Wayland. I didn't associate "X" with X11 :man-facepalming: . I fell so stupid. I'm so sorry for the confusion!

I've just tried the pdfpc apt package and have the same effect. Works when I set GDK_BACKEND=x11 and crashes otherwise.
